lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -43,118 +43,16 @@ def user_can_skip_entrance_exam(user, course):
     return False
 
 
-def user_has_passed_entrance_exam(request, course):
+def user_has_passed_entrance_exam(user, course):
     """
     Checks to see if the user has attained a sufficient score to pass the exam
     Begin by short-circuiting if the course does not have an entrance exam
     """
     if not course_has_entrance_exam(course):
         return True
-    if not request.user.is_authenticated():
```

Comment:
**Note to reviewers:** The changes in this file are essentially to prevent the code from repeatedly recomputing/fetching the grade to check whether the learner has passed the entrance exam.  Instead, a singular `None` response from `get_entrance_exam_content` indicates that the exam is passed.